stdlib/LinearAlgebra/test/eigen.jl

```diff
@@ -73,6 +73,16 @@ aimg  = randn(n,n)/2
             d,v = eigen(asym_sg, a_sg'a_sg)
             @test d == f.values
             @test v == f.vectors
+
+            # matrices of different types (#14896)
+            fs = eigen(Symmetric(asym_sg), a_sg'a_sg)
```

Comment:
You'd have to put this one (and the one in line 84) behind an `eltya <: Real` condition since it might be complex